comma/hypothesis.py

```diff
@@ -26,63 +30,312 @@ class Hypothesis:
+        # sort dataframe based on date of report
+        sorted_df = df_filtered.sort_values(by="Date_of_report", ascending=False)
+        # get the most recent data
+        agg_df = sorted_df.groupby("Date_of_statistics").first().reset_index()
+        daily_positive_cases = agg_df["Tested_positive"]
+
+        # match the length of the positives by day with the n of steps
+        if len(daily_positive_cases) < steps:
+            # if there are fewer data than steps
+            daily_positive_cases = cls.adjust_cases(steps, daily_positive_cases)
+        else:
+            # otherwise match the n of steps
+            # this prevents problems when there is more data than steps
+            daily_positive_cases = daily_positive_cases[:steps]
+        return daily_positive_cases
+
+    @staticmethod
+    def adjust_cases(steps: int, daily_positive_cases: pd.Series) -> pd.Series:
+        """
+        Ensures the length of daily_positive_cases matches the given steps.
+
+        If the length of daily_positive_cases is less than the provided steps,
+        (this might happen if there are missing data for some days in RIVM)
+        this function will repeat the last row of daily_positive_cases until
+        its length is equal to steps.
+
+        Args:
+            steps (int): Desired length for daily_positive_cases
+            daily_positive_cases (pd.Series): Series of positive cases per day
+
+        Returns:
+            pd.Series: Modified series with length equal to steps
+
+        """
+        n_times = steps - len(daily_positive_cases)
+
+        if n_times > 0:
+            n_values = pd.Series([daily_positive_cases.iloc[-1]] * n_times)
+
+            positive_cases = pd.concat(
+                [daily_positive_cases, n_values], ignore_index=True
+            )
+        return positive_cases
+
+    @staticmethod
+    def scale_cases_to_population(
+        daily_positive_cases: pd.Series, real_size: int, sim_size: int
+    ):
+        """
+        We compute the number of new positives in day _i_
+        for our simulated population as `new_cases_sim` = (n/N)*M where
+        N is the size of the simulated population, M is the size of the
+        real population, n is the number of new positives reported
+        on RIVM for that day.
+
+        Args:
+            daily_positive_cases(pd.Series): Daily positive cases
+            real_size(int): Size of the original population location
+            sim_size(int): Size of the simulated population
+
+        Returns:
+            pd.Series: Daily positive cases based on simulated population size
+        """
+        # Warning/ FYI:
+        # if sim_size is very small compared to real_size and
+        # the values in daily_positive_cases are also small
+        # (each value is much less than real_size), then the
+        # function could always result in zeros for every
+        # entry in daily_positive_cases.
```

Comment:
I think this comment should be part of the docstring.
Also, I am not sure how to read "could always result in zeros". Could it or does it always? Perhaps you can clarify the conditions under which if will result in zeros. 
```suggestion
        Warning/ FYI:
        if sim_size is very small compared to real_size and
        the values in daily_positive_cases are also small
        (each value is much less than real_size), then the
        function could always result in zeros for every
        entry in daily_positive_cases.
        """
```